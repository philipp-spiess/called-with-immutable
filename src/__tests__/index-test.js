import Immutable from "immutable";
import { overwriteDefaults } from "../index";
overwriteDefaults();

[
  "toBeCalledWithImmutable",
  "toHaveBeenCalledWithImmutable",
  "toBeCalledWith",
  "toHaveBeenCalledWith"
].forEach(calledWith => {
  ["jest.fn", "jasmine.createSpy"].forEach(mockName => {
    const getFunction = () => {
      return mockName === "jest.fn" ? jest.fn() : jasmine.createSpy("fn");
    };

    test(`${calledWith} works with ${mockName} and more expected arguments`, () => {
      const fn = getFunction();
      fn("foo");

      expect(fn).not[calledWith]("foo", "bar");

      expect(() =>
        expect(fn)[calledWith]("foo", "bar")
      ).toThrowErrorMatchingSnapshot();
    });

    test(`${calledWith} works with ${mockName} and more actual arguments`, () => {
      const fn = getFunction();
      fn("foo", "bar");

      expect(fn).not[calledWith]("foo");

      expect(() =>
        expect(fn)[calledWith]("foo")
      ).toThrowErrorMatchingSnapshot();
    });

    test(`${calledWith} works with ${mockName} and Immutable.js objects`, () => {
      const directlyCreated = new Immutable.Map([["foo", "bar"]]);
      const indirectlyCreated = new Immutable.Map().set("foo", "bar");
      const fn = getFunction();
      fn(directlyCreated, indirectlyCreated);

      expect(fn)[calledWith](indirectlyCreated, directlyCreated);

      expect(() =>
        expect(fn).not[calledWith](indirectlyCreated, directlyCreated)
      ).toThrowErrorMatchingSnapshot();
    });

    test(`${calledWith} works with ${mockName} and Immutable.js objects within regular objects`, () => {
      const directlyCreated = new Immutable.Map([["foo", "bar"]]);
      const indirectlyCreated = new Immutable.Map().set("foo", "bar");
      const fn = getFunction();
      fn({ a: directlyCreated }, { b: indirectlyCreated });

      expect(fn)[calledWith]({ a: indirectlyCreated }, { b: directlyCreated });

      expect(() =>
        expect(fn).not[calledWith](
          { a: indirectlyCreated },
          { b: directlyCreated }
        )
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
