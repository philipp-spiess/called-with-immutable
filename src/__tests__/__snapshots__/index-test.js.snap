// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeCalledWith works with jasmine.createSpy and Immutable.js objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toBeCalledWith(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toBeCalledWith works with jasmine.createSpy and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toBeCalledWith(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toBeCalledWith works with jasmine.createSpy and more actual arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toBeCalledWith(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toBeCalledWith works with jasmine.createSpy and more expected arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toBeCalledWith(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toBeCalledWith works with jest.fn and Immutable.js objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toBeCalledWith(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toBeCalledWith works with jest.fn and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toBeCalledWith(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toBeCalledWith works with jest.fn and more actual arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toBeCalledWith(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toBeCalledWith works with jest.fn and more expected arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toBeCalledWith(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toBeCalledWithImmutable works with jasmine.createSpy and Immutable.js objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toBeCalledWithImmutable(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toBeCalledWithImmutable works with jasmine.createSpy and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toBeCalledWithImmutable(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toBeCalledWithImmutable works with jasmine.createSpy and more actual arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toBeCalledWithImmutable(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toBeCalledWithImmutable works with jasmine.createSpy and more expected arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toBeCalledWithImmutable(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toBeCalledWithImmutable works with jest.fn and Immutable.js objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toBeCalledWithImmutable(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toBeCalledWithImmutable works with jest.fn and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toBeCalledWithImmutable(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toBeCalledWithImmutable works with jest.fn and more actual arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toBeCalledWithImmutable(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toBeCalledWithImmutable works with jest.fn and more expected arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toBeCalledWithImmutable(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toHaveBeenCalledWith works with jasmine.createSpy and Immutable.js objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toHaveBeenCalledWith(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toHaveBeenCalledWith works with jasmine.createSpy and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toHaveBeenCalledWith(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toHaveBeenCalledWith works with jasmine.createSpy and more actual arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toHaveBeenCalledWith(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toHaveBeenCalledWith works with jasmine.createSpy and more expected arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toHaveBeenCalledWith(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toHaveBeenCalledWith works with jest.fn and Immutable.js objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toHaveBeenCalledWith(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toHaveBeenCalledWith works with jest.fn and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toHaveBeenCalledWith(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toHaveBeenCalledWith works with jest.fn and more actual arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toHaveBeenCalledWith(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toHaveBeenCalledWith works with jest.fn and more expected arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toHaveBeenCalledWith(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jasmine.createSpy and Immutable.js objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jasmine.createSpy and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>spy</><dim>).not.toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected spy not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jasmine.createSpy and more actual arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jasmine.createSpy and more expected arguments 1`] = `
"<dim>expect(<red>spy</><dim>).toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected spy to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jest.fn and Immutable.js objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[Immutable.Map {foo: \\"bar\\"}, Immutable.Map {foo: \\"bar\\"}]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jest.fn and Immutable.js objects within regular objects 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).not.toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected mock function not to have been called with:
  <green>[{\\"a\\": Immutable.Map {foo: \\"bar\\"}}, {\\"b\\": Immutable.Map {foo: \\"bar\\"}}]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jest.fn and more actual arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\"]</>
But it was called with:
  <red>[\\"foo\\", \\"bar\\"]</>"
`;

exports[`toHaveBeenCalledWithImmutable works with jest.fn and more expected arguments 1`] = `
"<dim>expect(<red>jest.fn()</><dim>).toHaveBeenCalledWithImmutable(<green>expected</><dim>)

Expected mock function to have been called with:
  <green>[\\"foo\\", \\"bar\\"]</>
But it was called with:
  <red>[\\"foo\\"]</>"
`;
